### PR TITLE
Add shared block editing and plan review threads

### PR DESCRIPTION
**Pull request by steve8708**
## Summary
- add shared block registry/editor primitives and richer markdown collaboration support
- improve Agent-Native plan review with Figma-style comment authors, threaded replies, pinned/text comments, and email notifications
- update visual plan block rendering, wireframes, skill docs, template package wiring, and related changesets

## Tests
- pnpm --dir templates/plan exec vitest run app/pages/PlansPage.comments.test.ts server/lib/comment-notifications.spec.ts actions/update-visual-plan.spec.ts actions/get-plan-feedback.spec.ts server/plans.spec.ts server/plan-mdx.spec.ts
- pnpm --dir templates/plan typecheck
- pnpm --dir templates/plan build
- Playwright browser checks for pinned threads, compact comment popover, comments panel, and text-selection comments

### CODE REVIEW

File: templates/plan/actions/update-visual-plan.ts [hunk 1]
```diff
@@ -93,19 +100,19 @@ export default defineAction({
...
         );
       }
       if (isGuestAuthorIdentity(requesterEmail)) {
-        throw new Error("Commenting requires an account. Sign in to comment.");
+        throw new ForbiddenError(
+          "Commenting requires an account. Sign in to comment.",
+        );
       }
       const access = await resolveAccess("plan", args.planId);
       if (!access) throw new Error(`Plan ${args.planId} not found`);
```
[the remark this answers is not in the document]
Comment (steve8708) [reply templates/plan/actions/update-visual-plan.ts:128]: Fixed in 350d5b9d2. Comment-only writes now require a real requester identity (or local-mode owner fallback) before resolving plan access, so hosted unauthenticated callers and synthetic public/guest identities get a 403. Added regression coverage in `update-visual-plan.spec.ts`. Verified with focused plan tests/typecheck and full `pnpm run prep`.

File: templates/plan/app/pages/PlansPage.tsx [hunk 2]
```diff
@@ -911,6 +1717,8 @@ export function PlansPage() {
           message: buildCanvasMarkupFeedbackMessage(nextAnnotation),
           anchor: JSON.stringify(anchor),
           createdBy: "human",
+          authorEmail: collabUser?.email,
+          authorName: collabUser?.name,
         },
       ],
       note: "Human added canvas review markup.",
```
[the remark this answers is not in the document]
Comment (steve8708) [reply templates/plan/app/pages/PlansPage.tsx:1754]: Fixed in 350d5b9d2. Canvas text/callout markup tools are now gated to owner/admin/editor roles via `list-resource-shares`; reviewers still keep the comment pin flow, but cannot trigger content-patch markup paths. Added role coverage in `PlansPage.comments.test.ts`. Verified with focused plan tests/typecheck and full `pnpm run prep`.

File: templates/plan/actions/create-visual-plan.ts [hunk 3]
```diff
@@ -191,6 +195,13 @@ export default defineAction({
...
+            ...resolveCommentAuthor({
+              createdBy: comment.createdBy,
+              authorEmail: comment.authorEmail,
+              authorName: comment.authorName,
+              requestEmail: requesterEmail,
+              requestName: requesterName,
+            }),
             id: comment.id ?? newId("cmt"),
             planId: id,
             sectionId: comment.sectionId ?? null,
```
[the remark this answers is not in the document]
Comment (steve8708) [reply templates/plan/actions/create-visual-plan.ts]: Fixed in 350d5b9d2. `create-visual-plan` now preserves seeded `parentCommentId` links, copies parent section/kind/anchor context for replies, inserts parents before children, and rejects missing/cyclic parent references. Added `create-visual-plan.spec.ts` coverage for seeded threaded comments. Verified with focused tests/typecheck and full `pnpm run prep`.

File: templates/plan/server/db/schema.ts [hunk 4]
```diff
@@ -55,6 +55,7 @@ export const planComments = table("plan_comments", {
   planId: text("plan_id")
     .notNull()
     .references(() => plans.id),
+  parentCommentId: text("parent_comment_id"),
```
[the remark this answers is not in the document]
Comment (steve8708) [reply templates/plan/server/db/schema.ts]: Fixed in 350d5b9d2. `plan_comments.parent_comment_id` now references `plan_comments.id`, so orphaned parent references are constrained at the schema level. `templates/plan` typecheck and full `pnpm run prep` both pass locally.

**Comment by steve8708** [comment]
Addressed the Builder review from 2026-06-06 in 350d5b9d2:

- Required a real requester identity for hosted comment-only writes, while preserving local-mode comments.
- Hid and guarded canvas text/callout markup for non-editor reviewers; comment pins remain available.
- Preserved seeded `parentCommentId` threads in `create-visual-plan`, with parent ordering and missing/cycle validation.
- Added a self-reference FK on `plan_comments.parent_comment_id`.

Verification: focused plan tests/typecheck passed, and full `pnpm run prep` passed locally.

File: templates/plan/actions/create-visual-plan.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (steve8708) [reply templates/plan/actions/create-visual-plan.ts]: Fixed in 387c35503. create-visual-plan, create-ui-plan, and create-visual-questions now share insertInitialPlanComments, which preserves parentCommentId, inherits parent context for replies, inserts parents first, and rejects missing/cyclic parents with shared tests.

File: templates/plan/actions/update-visual-plan.ts [hunk 5]
```diff
@@ -93,19 +100,19 @@ export default defineAction({
...
           "Commenting on a plan requires an agent-native account. Sign in to leave a comment.",
         );
       }
       if (isGuestAuthorIdentity(requesterEmail)) {
-        throw new Error("Commenting requires an account. Sign in to comment.");
+        throw new ForbiddenError(
+          "Commenting requires an account. Sign in to comment.",
+        );
       }
       const access = await resolveAccess("plan", args.planId);
```
[the remark this answers is not in the document]
Comment (steve8708) [reply templates/plan/actions/update-visual-plan.ts:127]: Fixed in 387c35503. Comment-only writes now resolve the authenticated/local owner email before author attribution, while public anonymous viewers are still rejected.

File: templates/plan/app/pages/PlansPage.tsx [hunk 6]
```diff
@@ -911,6 +1717,8 @@ export function PlansPage() {
           message: buildCanvasMarkupFeedbackMessage(nextAnnotation),
```
[the remark this answers is not in the document]
Comment (steve8708) [reply templates/plan/app/pages/PlansPage.tsx:1747]: Fixed in 387c35503. Canvas markup tools are now gated to editor/admin/owner roles, while shared viewers can still leave normal comment pins.

File: templates/plan/server/db/schema.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (steve8708) [reply templates/plan/server/db/schema.ts]: Fixed in 387c35503. parentCommentId now has a self-referential foreign key on planComments.id, with seeded thread creation ordered parent-first.

**Comment by steve8708** [comment]
Follow-up in c9fb5bb36: the runtime plan DB migration now declares parent_comment_id with REFERENCES plan_comments(id), matching the Drizzle schema. Verified with a local SQLite FK syntax check, focused plan tests/typecheck, and full pnpm run prep.

**Comment by steve8708** [comment]
Addressed the final review edge cases in 2de373096:

- update-visual-plan now batches new comment inserts, resolves parents from both existing comments and pending rows, and inserts parent comments before replies.
- comment email notifications now receive a pre-insert comment snapshot, so each reply only notifies prior thread participants plus the plan owner; later replies in the same batch do not get notified by earlier replies.

Verification: focused plan comment tests passed, plan typecheck passed, pnpm install --frozen-lockfile passed, and full pnpm run prep passed locally.

File: templates/plan/actions/update-visual-plan.ts [hunk 7]
```diff
@@ -269,19 +288,50 @@ export default defineAction({
           continue;
         }
       }
+      const [parentComment] = comment.parentCommentId
+        ? await db
+            .select({
```
[the remark this answers is not in the document]
Comment (steve8708) [reply templates/plan/actions/update-visual-plan.ts]: Fixed in 2de373096. update-visual-plan now resolves new comment parents from both existing DB comments and the current batched insert set, orders parent rows before replies, and rejects missing or cyclic parent references. Added regression coverage for out-of-order batched threaded inserts and reran focused plan tests plus full prep locally.

**Comment by steve8708** [comment]
Addressed the latest Review Agent items from the 2026-06-06 01:45 UTC review:

- Comment-only requests that include an activity note now require editor access, so non-editor commenters cannot write arbitrary activity log messages.
- `update-visual-plan` now classifies and validates pending threaded comment inserts before plan, section, existing-comment, or new-comment writes, preventing partial persistence when batched comment validation fails.
- Added regression coverage for both cases.

Verification:
- `pnpm --dir templates/plan exec vitest run actions/update-visual-plan.spec.ts server/plans.spec.ts server/lib/comment-notifications.spec.ts --reporter=verbose`
- `pnpm --dir templates/plan typecheck`
- `pnpm run prep`

**Comment by steve8708** [comment]
Addressed the follow-up Review Agent item from the 2026-06-06 11:58 UTC review:

- Wrapped the visual plan write phase in a single `db.transaction`, covering the plan update, section upserts, existing comment updates, new comment inserts, consumed-comment updates, and the `plan.updated` event row.
- Kept validation/preflight work before the transaction, and kept post-commit reload, notifications, and local file writes after the transaction.
- Added regression coverage that the action persists updates and activity events through the transaction handle.

Verification:
- `pnpm --dir templates/plan exec vitest run actions/update-visual-plan.spec.ts server/plans.spec.ts server/lib/comment-notifications.spec.ts --reporter=verbose`
- `pnpm --dir templates/plan typecheck`
- `pnpm run prep` (passed with concurrent WIP files present in the working tree; only the two plan action files were committed)

**Comment by steve8708** [comment]
Addressed the 2026-06-06 12:12 UTC Review Agent regression:

- Comment-only requests may include the client-generated `note` again, so non-editor viewers can submit comments/replies.
- Comment-only activity events now ignore `args.note` for the event message and use the standard comment summary instead, so notes cannot spoof activity log messages.
- Updated the action regression test to cover a viewer comment with the client note.

Verification:
- `pnpm --dir templates/plan exec vitest run actions/update-visual-plan.spec.ts --reporter=verbose`

Note: broader local test runs are currently noisy because other agents have active uncommitted WIP in nearby plan/comment files; this commit only stages the two action files above.
